File: Difficulty: Easy/Missing in Array/missing-in-array.cpp
class Solution {
  public:
    int missingNum(vector<int>& arr) {
        // code here
        long long n = arr.size();

        long long total = (n + 1) * (n + 2) / 2;
        long long sum = 0;

        for (int x : arr) {
            sum += x;
        }

        return total - sum;
    }
};

File: Difficulty: Easy/Missing in Array/missing-in-array.java
class Solution {
    int missingNum(int arr[]) {
        // code here
        long n = arr.length + 1;
        long total = 1L*n*(n + 1)/2;
        long sum = 0;
        for (int x : arr) {
            sum += x;
        }
        return (int) (total - sum);
    }
}